hips/hip-9999.md

```diff
@@ -0,0 +1,92 @@
+---
+hip: "9999"
+title: "Add export-values directive (similar to import-values)"
+authors: [ "Jan von Loewenstein <jan.von.loewenstein@sap.com>", "Johannes Dillmann <j.dillmann@sap.com>", "Pavel Busko <pavel.busko@sap.com>", "Philipp Stehle <philipp.stehle@sap.com>", "Ralf Pannemans <ralf.pannemans@sap.com>", "Sumit Kulhadia <sumit.kulhadia@sap.com>" ]
+created: "2022-02-01"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+Composing Helm charts using dependencies is currently limited because it is not possible to create a chart with a defined set of values and distribute them to subcharts. Users of a parent chart instead have to provide values to the subcharts directly. Alternatively, users can provide `global` values, but in that case the subcharts need to be aware of the fact that they are used as a dependency. In addition, `global` values are available globally as the name suggests and hence multiple subcharts have to have aligned value keys. As a result, the parent chart and it's dependencies are rather tightly coupled.
+
+With the `import-values` directive, there is a mechanism in Helm to use the default values of a subchart as a value in the parent chart and define the name under which the subchart value is available in the parent chart.
+This HIP describes the `export-values` directive which is similar to the [`import-values`](https://helm.sh/docs/topics/charts/#importing-child-values-via-dependencies) directive, and allows exporting values from the parent chart to the subchart and defining the name under which it should be available.
+With that the values of the parent chart and the values of the subcharts can be defined independently. Subcharts can use their local values exclusively and don't have to be aware of the usage as a subchart. Parent charts on the other hand, can provide a consistent configuration api to their users and orchestrate the exchange of values between the subcharts using `export-values`, `import-values` or a combination thereof.
+
+This change was discussed in [helm/#3242]. There were several attempts to implement this: [helm/#7477], [helm/#10059].
+
+## Specification
+
+With the proposed `export-values` directive, useres are able to specify in the **parent** chart (snippet below) which and where values should be exported to the **subchart**.
+
+```yaml
+# Chart.yaml:
+dependencies:
+  - name: client
+    version: 1.0.0
+    export-values:
+    - parent: "port"
+      child: "port"
+  - name: server
+    version: 1.0.0
+    export-values:
+    - parent: "port"
+      child: "port"
```

Comment:
The [import-values]() directive has two different formats. 

The [exports format](https://v2.helm.sh/docs/developing_charts/#using-the-exports-format) that relies on the convention that child charts put values under an `exports` key in their values files and a jsonpath like syntaxt to define the target of the export in the parent's value structure.

The [child-parent format](https://v2.helm.sh/docs/developing_charts/#using-the-child-parent-format) that has a jsonpath like syntax for both pointing to the exports and the target in the parent's value structure.


We seem to be proposing the latter and I don't know if the former would make any sense or be possible at all. However, we definitely need to reference these options for `import-values` and present an argument for our choice.